codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/generators/ServerOperationRegistryGeneration.kt

```diff
@@ -0,0 +1,281 @@
+     */
+    private fun renderOperationRegistryStruct(writer: RustWriter) {
+        // A lot of things can become pretty complex in this type as it will hold 2 generics per operation
+        Attribute.Custom("allow(clippy::all)").render(writer)
+        writer.rustBlock(
+            """
+            pub struct $operationRegistryNameWithArguments
+            """.trimIndent()
+        ) {
+            val members = operationNames
+                .mapIndexed { i, operationName -> "$operationName: Op$i" }
+                .joinToString(separator = ",\n")
+            rustTemplate(
+                """
+                $members,
+                _phantom: #{Phantom}<(B, ${phantomMembers()})>,
+                """,
+                *codegenScope
+            )
+        }
+    }
+
+    /*
+     * Renders the OperationRegistryBuilder structure, used to build the OperationRegistry and then convert it
+     * into a Smithy Router.
+     */
+    private fun renderOperationRegistryBuilderStruct(writer: RustWriter) {
+        // A lot of things can become pretty complex in this type as it will hold 2 generics per operation
+        Attribute.Custom("allow(clippy::all)").render(writer)
+        writer.rustBlock(
+            """
+            pub struct $operationRegistryBuilderNameWithArguments
+            """.trimIndent()
+        ) {
+            val members = operationNames
+                .mapIndexed { i, operationName -> "$operationName: Option<Op$i>" }
+                .joinToString(separator = ",\n")
+            rustTemplate(
+                """
+                $members,
+                _phantom: #{Phantom}<(B, ${phantomMembers()})>,
+                """,
+                *codegenScope
+            )
+        }
+    }
+
+    /*
+     * Renders the OperationRegistryBuilder Error type, used to error out in case there are uninitialized fields.
+     * This structs implement Debug, Display and std::error::Error.
+     */
+    private fun renderOperationRegistryBuilderError(writer: RustWriter) {
+        // derive[Debug] is needed to impl std::error::Error
+        Attribute.Derives(setOf(RuntimeType.Debug)).render(writer)
+        writer.rustTemplate(
+            """
+            pub enum ${operationRegistryBuilderName}Error {
+                UninitializedField(&'static str)
+            }
+            impl std::fmt::Display for ${operationRegistryBuilderName}Error {
+                fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+                    match self {
+                        Self::UninitializedField(v) => write!(f, "{}", v),
+                    }
+                }
+            }
+            impl #{StdError} for ${operationRegistryBuilderName}Error {}
+            """.trimIndent(),
+            *codegenScope
+        )
+    }
+
+    /*
+     * Renders the OperationRegistryBuilder Default implementation , used to create a new builder that can be
+     * later filled with operations and their routees.
```

Comment:
```suggestion
     * Renders the `OperationRegistryBuilder` `Default` implementation, used to create a new builder that can be
     * populated with the service's operations.
```

The user does not define the routes.